packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

```diff
@@ -146,39 +72,287 @@ const TODO = (method: keyof Client) => {
+                    : [])
+            ]
+        }
+    },
+    addPaymentInstrumentToBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
     addPriceBooksToBasket: TODO('addPriceBooksToBasket'),
     addTaxesForBasket: TODO('addTaxesForBasket'),
     addTaxesForBasketItem: TODO('addTaxesForBasketItem'),
-    createBasket: updateBasketWithResponseBasketId,
+    createBasket(customerId, {parameters}, response) {
+        const {basketId} = response
+
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId && basketId
+                    ? [
+                          {
+                              // @ts-ignore
+                              queryKey: getCustomerBaskets.queryKey({
+                                  ...parameters,
+                                  basketId,
+                                  customerId
+                              }),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(
+                                      {...parameters, basketId},
+                                      response,
+                                      oldData
+                                  )
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
     createShipmentForBasket: TODO('createShipmentForBasket'),
-    deleteBasket: (customerId, {parameters}) => ({
-        // TODO: Convert invalidate to an update that removes the matching basket
-        ...invalidateCustomerBasketsQuery(customerId, parameters),
-        remove: [
-            and(
-                matchParameters(toGetBasketParameters(parameters)),
-                matchesPath(getBasketPath(parameters))
-            )
-        ]
-    }),
-    mergeBasket: updateBasketWithResponseBasketId,
-    removeCouponFromBasket: updateBasket,
+    deleteBasket(customerId, {parameters}) {
+        return {
+            // TODO: Convert invalidate to an update that removes the matching basket
+            ...invalidateCustomerBasketsQuery(customerId, parameters),
+            remove: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [{queryKey: getCustomerBaskets.queryKey({...parameters, customerId})}]
+                    : [])
```

Comment:
I don't think we want to remove _all_ baskets, right? Just the one that got deleted? So just invalidating the query should be sufficient. (Or, if you want to be fancy, you could do an `update` that removes the matching basket from the `baskets` array.)

Reply:
Makes sense to only invalidating the Shopper Customer basket.

I changed the `remove` to do a fuzzy match to remove all the cache entries containing the  in their queryKey `path`. That will remove from cache all the rest of basket related endpoints like: `'/payment-methods'`, `'/price-books'`, etc.